src/github.com/getlantern/balancer/balancer.go

```diff
@@ -22,23 +22,42 @@ var (
 // Balancer balances connections established by one or more Dialers.
 type Balancer struct {
 	dialers []*dialer
+	trusted []*dialer
 }
 
 // New creates a new Balancer using the supplied Dialers.
 func New(dialers ...*Dialer) *Balancer {
 	dhs := make([]*dialer, 0, len(dialers))
+
 	for _, d := range dialers {
 		dl := &dialer{Dialer: d}
 		dl.start()
 		dhs = append(dhs, dl)
 	}
+
 	// Sort dialers by QOS (ascending) for later selection
 	sort.Sort(byQOSAscending(dhs))
+
 	return &Balancer{
 		dialers: dhs,
 	}
 }
 
+// trustedDialers returns the subset of b.dialers that are considered as
+// trusted.
+func (b *Balancer) trustedDialers() []*dialer {
+	if b.trusted == nil {
+		b.trusted = make([]*dialer, 0, len(b.dialers))
+		// Lazy initialization of trusted dialers.
```

Comment:
There is no synchronisation here, so if a second goroutine calls `trustedDialers` while the first call is at this point, they'll get an initially empty `b.trusted`, which will get filled afterwards while they're possibly using it.